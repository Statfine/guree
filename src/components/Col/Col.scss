@import "../../common/utils/index.scss";

$total: 24;

@function columnOffset($col, $total: $total) {
  @if $col == 0 {
    @return 0;
  } @else {
    @return percentage($col/$total);
  }
}

@mixin make-grid($responsive: "") {
  @for $i from 0 to $total + 1 {
    @if $i == 0 {
      &-#{$responsive}offset-0 {
        margin-left: 0;
      }
      &-#{$responsive}order-0 {
        order: 0;
      }
      &-push-0 {
        left: auto;
      }

      &-pull-0 {
        right: auto;
      }
      &-0 {
        display: none;
      }
    } @else {
      &-#{$responsive}-offset-#{$i} {
        margin-left: columnOffset($i);
      }
      &-#{$responsive}-order-#{$i} {
        order: $i;
      }
      &-#{$responsive}-push-#{$i} {
        left: columnOffset($i);
      }
      &-#{$responsive}-pull-#{$i} {
        right: columnOffset($i);
      }
      &-#{$responsive}-#{$i} {
        display: block;
        flex: 0 0 columnOffset($i);
        max-width: columnOffset($i);
      }
    }
  }
}

.#{scopedClass(col)} {
  max-width: 100%;
  min-height: 1px;
  position: relative;
  @include make-grid();
  // xs
  @media only screen and (max-width: 576px) {
    @include make-grid("xs");
  }
  // sm
  @media only screen and (min-width: 576px) {
    @include make-grid("sm");
  }
  // md
  @media only screen and (min-width: 768px) {
    @include make-grid("md");
  }
  // lg
  @media only screen and (min-width: 992px) {
    @include make-grid("lg");
  }
  // xl
  @media only screen and (min-width: 1200px) {
    @include make-grid("xl");
  }
  // xxl
  @media only screen and (min-width: 1600px) {
    @include make-grid("xxl");
  }
}
