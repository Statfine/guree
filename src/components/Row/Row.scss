@import "../../common/utils/index.scss";

.#{scopedClass(row)} {
  display: flex;
  flex-flow: row wrap;

  &-top {
    align-items: flex-start;
  }
  &-middle {
    align-items: center;
  }
  &-bottom {
    align-items: flex-end;
  }
  &-start {
    justify-content: flex-start;
  }
  &-end {
    justify-content: flex-end;
  }
  &-center {
    justify-content: center;
  }
  &-space-around {
    justify-content: space-around;
  }
  &-space-between {
    justify-content: space-between;
  }
}
