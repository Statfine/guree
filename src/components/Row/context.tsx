import { createContext, Context } from "react";

export interface RowContextState {
  horizontal?: number;
  vertical?: number;
}

const RowContext: Context<RowContextState> = createContext({});

export default RowContext;
