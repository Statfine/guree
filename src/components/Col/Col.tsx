import React, { CSSProperties } from "react";
import "./Col.scss";
import { combineClasses, scopedClass, classesObj } from "@utils/index";
import RowContext from "@components/Row/context";

type StringOrNumber = string | number;
type ResponsiveAttributeType = StringOrNumber | Object;

interface ColSize extends Object {
  offset?: StringOrNumber;
  order?: StringOrNumber;
  push?: StringOrNumber;
  pull?: StringOrNumber;
  span?: StringOrNumber;
}

export interface Props {
  children?: React.ReactNode;
  /** flex 布局属性 */
  flex?: StringOrNumber;
  /** 栅格左侧的间隔格数 */
  offset?: StringOrNumber;
  /** 栅格顺序 */
  order?: StringOrNumber;
  /** 栅格向右移动格数 */
  push?: StringOrNumber;
  /** 栅格向左移动格数 */
  pull?: StringOrNumber;
  /** 栅格占位格数，为 0 时等于 display: none */
  span?: StringOrNumber;
  /** <576px 响应式栅格，可为栅格数或一个包含其他属性的对象 */
  xs?: ResponsiveAttributeType;
  /** ≥576px 响应式栅格，可为栅格数或一个包含其他属性的对象 */
  sm?: ResponsiveAttributeType;
  /** ≥768px 响应式栅格，可为栅格数或一个包含其他属性的对象 */
  md?: ResponsiveAttributeType;
  /** ≥992px 响应式栅格，可为栅格数或一个包含其他属性的对象 */
  lg?: ResponsiveAttributeType;
  /** ≥1200px 响应式栅格，可为栅格数或一个包含其他属性的对象	 */
  xl?: ResponsiveAttributeType;
  /** ≥1600px 响应式栅格，可为栅格数或一个包含其他属性的对象 */
  xxl?: ResponsiveAttributeType;
}

const Col: React.FunctionComponent<Props> = ({
  children,
  flex,
  offset = 0,
  order = 0,
  push = 0,
  pull = 0,
  span,
  xs,
  sm,
  md,
  lg,
  xl,
  xxl,
}: Props) => {
  const flexStyle: CSSProperties = {
    flex,
  };

  const colClass = (...classes: (string | Array<string> | classesObj)[]) =>
    scopedClass("col", ...classes);

  const generateReponsiveClassNames = (
    attributeName: string,
    attributeValue: ResponsiveAttributeType
  ) => {
    if (
      typeof attributeValue === "string" ||
      typeof attributeValue === "number"
    ) {
      return {
        [`${colClass(attributeName, `${attributeValue}`)}`]: Boolean(
          attributeValue
        ),
      };
    } else {
      // typeof attributeValue => "object"
      const { offset, order, push, pull, span } = attributeValue as ColSize;
      return {
        [`${colClass(`${attributeName}-offset`, `${offset}`)}`]: true,
        [`${colClass(`${attributeName}-order`, `${order}`)}`]: true,
        [`${colClass(`${attributeName}-push`, `${push}`)}`]: true,
        [`${colClass(`${attributeName}-pull`, `${pull}`)}`]: true,
        [`${colClass(`${attributeName}`, `${span}`)}`]: true,
      };
    }
  };

  const generateReponsiveClassNames2 = (obj: {
    [k: string]: ResponsiveAttributeType | undefined;
  }) => {
    return Object.entries(obj).map(([key, value]) => {
      if (value === undefined) {
        return "";
      }
      return generateReponsiveClassNames(key, value);
    });
  };

  const reponsiveClassNames = generateReponsiveClassNames2({
    xs,
    sm,
    md,
    lg,
    xl,
    xxl,
  });
  
  const styleClassNames = (offset: StringOrNumber, order: StringOrNumber) =>{
    return combineClasses(
      {
        [`${colClass("offset", `${offset}`)}`]: Boolean(offset),
        [`${colClass("order", `${order}`)}`]: Boolean(order),
        [`${colClass("push", `${push}`)}`]: Boolean(push),
        [`${colClass("pull", `${pull}`)}`]: Boolean(pull),
      },
      (typeof span === "undefined")
        ? {}
        : {
          [`${colClass(`${span}`)}`]: true,
        },
      ...reponsiveClassNames
    );
  }
    
  const className = combineClasses(colClass(), styleClassNames(offset, order));

  return (
    <RowContext.Consumer>
      {({ horizontal, vertical }) => {
        const gutterStyle: CSSProperties = {
          ...(horizontal && horizontal > 0
            ? {
                paddingLeft: horizontal / 2,
                paddingRight: horizontal / 2,
              }
            : {}),
          ...(vertical && vertical! > 0
            ? {
                paddingTop: vertical / 2,
                paddingBottom: vertical / 2,
              }
            : {}),
        };
        return (
          <div className={className} style={{ ...flexStyle, ...gutterStyle }}>
            {children}
          </div>
        );
      }}
    </RowContext.Consumer>
  );
};

export default Col;
