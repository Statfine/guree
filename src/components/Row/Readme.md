```jsx
import Col from "@components/Col";
<Row justify="start">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="end">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="center">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="space-around">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="space-between">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row justify="space-between">
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
  <Col span={6}>col-6</Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row gutter={16}>
  <Col className="gutter-row" span={6}>
    <div style={{ background: "#00a0e9" }}>col-6</div>
  </Col>
  <Col className="gutter-row" span={6}>
    <div style={{ background: "#00a0e9" }}>col-6</div>
  </Col>
  <Col className="gutter-row" span={6}>
    <div style={{ background: "#00a0e9" }}>col-6</div>
  </Col>
  <Col className="gutter-row" span={6}>
    <div style={{ background: "#00a0e9" }}>col-6</div>
  </Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row gutter={{ xs: 8, sm: 16, md: 24, lg: 32 }}>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row gutter={[16, 24]}>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
</Row>;
```

```jsx
import Col from "@components/Col";
<Row
  gutter={[
    { xs: 2, sm: 4, md: 6, lg: 8, xl: 10, xxl: 12 },
    { xs: 14, sm: 16, md: 18, lg: 20, xl: 22, xxl: 24 },
  ]}
>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
  <Col className="gutter-row" span={6}>
    col-6
  </Col>
</Row>;
```
