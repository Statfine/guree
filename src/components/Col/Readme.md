### 基本用法

```jsx
import Row from "@components/Row";
<div className="grid-demo">
  <Row>
    <Col span={4}></Col>
    <Col span={4}></Col>
    <Col span={4}></Col>
    <Col span={4}></Col>
    <Col span={4}></Col>
    <Col span={4}></Col>
  </Row>
</div>;
```

### flex 布局

```jsx
import Row from "@components/Row";
<div className="grid-demo">
  <Row>
    <Col flex={2}></Col>
    <Col flex="100px"></Col>
    <Col flex="auto"></Col>
  </Row>
</div>;
```

### 左侧间隔

```jsx
import Row from "@components/Row";
<div className="grid-demo">
  <Row>
    <Col span={6}></Col>
    <Col span={6}></Col>
    <Col span={4} offset={2}></Col>
    <Col span={6}></Col>
  </Row>
</div>;
```

### 排列顺序

```jsx
import Row from "@components/Row";
<div className="grid-demo">
  <Row>
    <Col span={6}>1</Col>
    <Col span={6}>2</Col>
    <Col span={6} order={2}>
      3
    </Col>
    <Col span={6} order={1}>
      4
    </Col>
  </Row>
</div>;
```

### 左右偏移

```jsx
import Row from "@components/Row";
<div className="grid-demo">
  <Row>
    <Col span={6} push={18}></Col>
    <Col span={18} pull={6}></Col>
  </Row>
</div>;
```

### 响应式布局

```jsx
import Row from "@components/Row";
<div className="grid-demo">
  <Row justify="space-around">
    <Col span={4} xs={6} sm={5} md={4} lg={3} xl={2} xxl={1}></Col>
    <Col span={4} xs={6} sm={5} md={4} lg={3} xl={2} xxl={1}></Col>
    <Col span={4} xs={6} sm={5} md={4} lg={3} xl={2} xxl={1}></Col>
    <Col span={4} xs={6} sm={5} md={4} lg={3} xl={2} xxl={1}></Col>
  </Row>
</div>;
```
