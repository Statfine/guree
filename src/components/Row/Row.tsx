import React, { ReactNode, useState, CSSProperties } from "react";
import "./Row.scss";
import { combineClasses, scopedClass, classesObj } from "@utils/index";
import RowContext from "@components/Row/context";

type GutterType =
  | number
  | {
      xs?: number;
      sm?: number;
      md?: number;
      lg?: number;
      xl?: number;
      xxl?: number;
    };

interface Props {
  children?: ReactNode;
  align?: "top" | "middle" | "bottom";
  justify?: "start" | "end" | "center" | "space-around" | "space-between";
  gutter?: GutterType | [GutterType, GutterType];
}

const Row: React.FunctionComponent<Props> = ({
  children,
  align,
  justify,
  gutter,
}: Props) => {
  const colClass = (...classes: (string | Array<string> | classesObj)[]) =>
    scopedClass("row", ...classes);

  const MediaMap: {
    [k: string]: string;
  } = {
    xs: "(max-width: 576px)",
    sm: "(min-width: 576px)",
    md: "(min-width: 768px)",
    lg: "(min-width: 992px)",
    xl: "(min-width: 1200px)",
    xxl: "(min-width: 1600px)",
  };

  const parseGutter = (gutter: GutterType) => {
    if (typeof gutter === "number") {
      return gutter;
    }
    return Object.entries(gutter).reduce((res, cur) => {
      const [mediaKey, mediaValue] = cur;
      const mql = window.matchMedia(MediaMap[mediaKey]);
      if (mql.matches) {
        return mediaValue || 0;
      }
      return res;
    }, 0);
  };

  const getGutter = (
    gutter: GutterType | [GutterType, GutterType] | undefined
  ) => {
    if (typeof gutter === "undefined") {
      return {
        horizontal: 0,
        vertical: 0,
      };
    }
    if (Array.isArray(gutter)) {
      return {
        horizontal: parseGutter(gutter[0]),
        vertical: parseGutter(gutter[1]),
      };
    }
    return {
      horizontal: parseGutter(gutter),
      vertical: 0,
    };
  };

  const [{ horizontal, vertical }, setState] = useState(getGutter(gutter));

  const responiveGutter = (gutter: GutterType, cb: Function, key: string) => {
    if (typeof gutter === "number") {
      return gutter;
    }
    Object.entries(gutter).forEach(([mediaKey, mediaValue]) => {
      const mql = window.matchMedia(MediaMap[mediaKey]);
      mql.addListener((e) => {
        if (e.matches) {
          cb({
            [key]: mediaValue || 0,
          });
        }
      });
    });
  };

  const bindGutter = (
    gutter: GutterType | [GutterType, GutterType] | undefined,
    setState: React.Dispatch<
      React.SetStateAction<{
        horizontal: number;
        vertical: number;
      }>
    >
  ) => {
    if (typeof gutter === "undefined") {
      return;
    }
    if (Array.isArray(gutter)) {
      gutter.forEach((x, i) => {
        responiveGutter(x, setState, ["horizontal", "vertical"][i]);
      });
    } else {
      responiveGutter(gutter, setState, "horizontal");
    }
  };

  bindGutter(gutter, setState);

  const styleClassNames = (align?: string, justify?: string) => {
    return combineClasses({
      [`${colClass(`${align}`)}`]: Boolean(align),
      [`${colClass(`${justify}`)}`]: Boolean(justify),
    });
  };

  const gutterStyle: CSSProperties = {
    ...(horizontal > 0
      ? {
          marginLeft: horizontal / -2,
          marginRight: horizontal / -2,
        }
      : {}),
    ...(vertical! > 0
      ? {
          marginTop: vertical / -2,
          marginBottom: vertical / 2,
        }
      : {}),
  };

  const className = combineClasses(colClass(), styleClassNames(align, justify));
  return (
    <RowContext.Provider value={{ horizontal, vertical }}>
      <div className={className} style={gutterStyle}>
        {children}
      </div>
    </RowContext.Provider>
  );
};

export default Row;
