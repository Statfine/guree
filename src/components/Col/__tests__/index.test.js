import React from "react";
import renderer from "react-test-renderer";
import Col from "..";
import Row from "@components/Row";

describe("Col", () => {
  it("正确导出 Col 组件", () => {
    const col = renderer
      .create(
        <Row>
          <Col span={4}></Col>
        </Row>
      )
      .toJSON();
    expect(col).toMatchSnapshot();
  });

  it("测试 Col 组件 flex 参数", () => {
    const col = renderer
      .create(
        <Row>
          <Col flex="1 1"></Col>
          <Col flex={1}></Col>
        </Row>
      )
      .toJSON();
    expect(col).toMatchSnapshot();
  });

  it("测试 Col 组件 offset 参数", () => {
    const col = renderer
      .create(
        <Row>
          <Col span={4} offset="1"></Col>
          <Col span={4} offset={1}></Col>
        </Row>
      )
      .toJSON();
    expect(col).toMatchSnapshot();
  });
  it("测试 Col 组件 order 参数", () => {
    const col = renderer
      .create(
        <Row>
          <Col span={4} order="1"></Col>
          <Col span={4} order={1}></Col>
        </Row>
      )
      .toJSON();
    expect(col).toMatchSnapshot();
  });
  it("测试 Col 组件 push 参数", () => {
    const col = renderer
      .create(
        <Row>
          <Col span={4} push="1"></Col>
          <Col span={4} push={1}></Col>
        </Row>
      )
      .toJSON();
    expect(col).toMatchSnapshot();
  });
  it("测试 Col 组件 offset 参数", () => {
    const col = renderer
      .create(
        <Row>
          <Col span={4} offset="1"></Col>
          <Col span={4} offset={1}></Col>
        </Row>
      )
      .toJSON();
    expect(col).toMatchSnapshot();
  });

  it("测试 Col 组件 xs sm md lg xl xxl 参数", () => {
    const col = renderer
      .create(
        <Row>
          <Col span={4} xs={1} sm={2} md={3} lg={4} xl={5} xxl={6}></Col>
          <Col span={4} xs="1" sm="2" md="3" lg="4" xl="5" xxl="6"></Col>
          <Col
            span={4}
            xs={{ offset: 1, order: 1, push: 1, pull: 1, span: 1 }}
          ></Col>{" "}
          <Col
            span={4}
            sm={{ offset: 1, order: 1, push: 1, pull: 1, span: 1 }}
          ></Col>
          <Col
            span={4}
            md={{ offset: 1, order: 1, push: 1, pull: 1, span: 1 }}
          ></Col>
          <Col
            span={4}
            lg={{ offset: 1, order: 1, push: 1, pull: 1, span: 1 }}
          ></Col>
          <Col
            span={4}
            xl={{ offset: 1, order: 1, push: 1, pull: 1, span: 1 }}
          ></Col>
          <Col
            span={4}
            xxl={{ offset: 1, order: 1, push: 1, pull: 1, span: 1 }}
          ></Col>
        </Row>
      )
      .toJSON();
    expect(col).toMatchSnapshot();
  });
});
