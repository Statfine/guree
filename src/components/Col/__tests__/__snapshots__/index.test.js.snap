// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col 正确导出 Col 组件 1`] = `
<div
  className="guree-row"
  style={Object {}}
>
  <div
    className="guree-col guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
</div>
`;

exports[`Col 测试 Col 组件 flex 参数 1`] = `
<div
  className="guree-row"
  style={Object {}}
>
  <div
    className="guree-col"
    style={
      Object {
        "flex": "1 1",
      }
    }
  />
  <div
    className="guree-col"
    style={
      Object {
        "flex": 1,
      }
    }
  />
</div>
`;

exports[`Col 测试 Col 组件 offset 参数 1`] = `
<div
  className="guree-row"
  style={Object {}}
>
  <div
    className="guree-col guree-col-offset-1 guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-offset-1 guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
</div>
`;

exports[`Col 测试 Col 组件 offset 参数 2`] = `
<div
  className="guree-row"
  style={Object {}}
>
  <div
    className="guree-col guree-col-offset-1 guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-offset-1 guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
</div>
`;

exports[`Col 测试 Col 组件 order 参数 1`] = `
<div
  className="guree-row"
  style={Object {}}
>
  <div
    className="guree-col guree-col-order-1 guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-order-1 guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
</div>
`;

exports[`Col 测试 Col 组件 push 参数 1`] = `
<div
  className="guree-row"
  style={Object {}}
>
  <div
    className="guree-col guree-col-push-1 guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-push-1 guree-col-4"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
</div>
`;

exports[`Col 测试 Col 组件 xs sm md lg xl xxl 参数 1`] = `
<div
  className="guree-row"
  style={Object {}}
>
  <div
    className="guree-col guree-col-4 guree-col-xs-1 guree-col-sm-2 guree-col-md-3 guree-col-lg-4 guree-col-xl-5 guree-col-xxl-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-4 guree-col-xs-1 guree-col-sm-2 guree-col-md-3 guree-col-lg-4 guree-col-xl-5 guree-col-xxl-6"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-4 guree-col-xs-offset-1 guree-col-xs-order-1 guree-col-xs-push-1 guree-col-xs-pull-1 guree-col-xs-1"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
   
  <div
    className="guree-col guree-col-4 guree-col-sm-offset-1 guree-col-sm-order-1 guree-col-sm-push-1 guree-col-sm-pull-1 guree-col-sm-1"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-4 guree-col-md-offset-1 guree-col-md-order-1 guree-col-md-push-1 guree-col-md-pull-1 guree-col-md-1"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-4 guree-col-lg-offset-1 guree-col-lg-order-1 guree-col-lg-push-1 guree-col-lg-pull-1 guree-col-lg-1"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-4 guree-col-xl-offset-1 guree-col-xl-order-1 guree-col-xl-push-1 guree-col-xl-pull-1 guree-col-xl-1"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
  <div
    className="guree-col guree-col-4 guree-col-xxl-offset-1 guree-col-xxl-order-1 guree-col-xxl-push-1 guree-col-xxl-pull-1 guree-col-xxl-1"
    style={
      Object {
        "flex": undefined,
      }
    }
  />
</div>
`;
